### PR TITLE
SP2-903 design review remediation

### PR DESCRIPTION
**Pull request by alexw-moj**
Now adds ‘This is a future goal.’ in place of the ‘Aim to achieve…’ text.
When there are no steps: should say ‘No steps added.’ Next to a link that says ‘Add steps.’
Uses the new `status` params.

RemoveGoal page should now return to the correct page, depending on the chosen path. 
- `Update -> Remove a goal -> Do not remove a goal` should now go back to `Update` instead of `Plan`
- `Remove -> Do not remove a goal` goes to `Plan`

Now selects the removed goals tab after removing a goal.

### CODE REVIEW

File: server/routes/createGoal/CreateGoalController.ts [hunk 1]
```diff
@@ -50,6 +50,9 @@ export default class CreateGoalController {
     const selectedAreaOfNeed = areasOfNeed.find(areaOfNeed => areaOfNeed.url === req.params.areaOfNeed)
     const minimumDatePickerDate = formatDateWithStyle(new Date().toISOString(), 'short')
 
+    // TODO: How many places can I get here from?
```
Comment (pipwilson) [inline server/routes/createGoal/CreateGoalController.ts]: not quite sure what these TODOs are meant to be suggesting. If there's something concrete here TODO in the future, can you be specific?

File: server/views/pages/update-goal.njk [hunk 2]
```diff
@@ -113,52 +113,52 @@
                         </tbody>
                     </table>
                 {% else %}
-                    <div class="goal-summary-card__steps--empty">
-                        <strong>{{ locale.stepsList.noStepsAdded }}</strong>
-                        {% if data.actions.length > 0 %}
-                            <a href="/goal/{{ data.goal.uuid }}/add-steps">{{ locale.stepsList.addSteps }}</a>
-                        {% endif %}
+                    <div class="govuk-summary-card govuk-summary-card--no-pad">
```
Comment (pipwilson) [inline server/views/pages/update-goal.njk]: The markup here is different to that in goal-summary-card which displays when there are no steps - can you compare and consolidate to one approach please?

File: server/routes/createGoal/CreateGoalController.ts (hunk 1 above)
Comment (alexw-moj) [reply server/routes/createGoal/CreateGoalController.ts]: Removed in https://github.com/ministryofjustice/hmpps-sentence-plan-ui/pull/186/commits/2a84f8fc79e24634664948ecb231cd94aa9de143

File: server/views/pages/update-goal.njk (hunk 2 above)
Comment (alexw-moj) [reply server/views/pages/update-goal.njk]: Updated in https://github.com/ministryofjustice/hmpps-sentence-plan-ui/pull/186/commits/2a84f8fc79e24634664948ecb231cd94aa9de143

File: assets/scss/components/_goal-summary-card.scss [hunk 3]
```diff
@@ -37,6 +37,13 @@
       padding-bottom: govuk-spacing(2);
       box-shadow: inset 0 -1px $govuk-border-colour;
     }
+
+    &--empty-no-shadow {
```
Comment (pipwilson) [inline assets/scss/components/_goal-summary-card.scss:41]: OK I'll bite - do you want to show me the difference between these two uses tomorrow?

Comment (alexw-moj) [reply assets/scss/components/_goal-summary-card.scss:41]: Will do.

**Review by pipwilson — APPROVED** [review]
LGTM

**Pull request merged**